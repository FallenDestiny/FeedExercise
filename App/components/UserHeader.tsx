import React, { ReactElement } from 'react'
import { View, Image, Text, ViewStyle, TouchableOpacity, StyleSheet } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import { colors } from '../utils/colors'
import fonts from '../assets/fonts'
import { UserData } from '../redux/App/actions'
var moment = require('moment');

interface Props {
    user: UserData,
    date?: Date,
    style?: ViewStyle,
    onPress: () => void,
    testID? : string
}

export default function UserHeader({ user, date, style, onPress, testID }: Props): ReactElement {
    return (
        <TouchableOpacity testID={testID} activeOpacity={0.75} style={[styles.header, style]} onPress={onPress}>
            {
                user && user.avatar ?
                    <Image style={styles.userAvatar} source={{ uri: user.avatar }} /> :
                    null
            }

            <View>
                {
                    user && user.name ?
                        <Text style={styles.userTitle}>{user.name}</Text>
                        : null
                }
                {
                    date ?
                        <Text style={[styles.userTitle, styles.createdStyle]}>{moment(date).fromNow()}</Text>
                        : null
                }

            </View>
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    header: {
        flex: 1,
        maxHeight: 55,
        flexDirection: 'row',
        alignItems: 'center',
    },
    userAvatar: {
        aspectRatio: 1,
        height: '100%',
        borderRadius: 27.5,
        resizeMode: 'contain',
    },
    userTitle: {
        color: colors.color1,
        fontFamily: fonts.sf_pro_display_bold,
        fontSize: 13,
        marginLeft: 15
    },
    createdStyle: {
        color: colors.color2
    },
})