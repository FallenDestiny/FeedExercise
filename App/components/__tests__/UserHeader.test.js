import React from 'react';
import renderer from 'react-test-renderer';
import UserHeader from '../UserHeader'

const date = "2020-02-18T06:33:57.162Z"
const user = {
    "id": "1",
    "postId": "1",
    "last_seen": "2020-02-18T06:33:57.162Z",
    "name": "Charles Hickle",
    "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/bruno_mart/128.jpg",
    "followers": 10,
    "followings": 53,
    "description": "Extended logistical array",
    "posts": [
        {
            "id": "1",
            "createdAt": "2020-02-18T04:40:49.601Z",
            "title": "Principal Identity Associate",
            "image": "https://images.unsplash.com/photo-1582043725042-f3d1873eeadf?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80",
            "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
            "likes": 68,
            "comments": 43
        }
    ]
}

describe('UserHeader', () => {
    describe('Rendering', () => {
        it('draw all the header data with date', () => {

            const component = renderer.create(<UserHeader user={user} date={date}/>).toJSON()
            expect(component).toMatchSnapshot()
        });
        it('draw all the header data without date', () => {

            const component = renderer.create(<UserHeader user={user}/>).toJSON()
            expect(component).toMatchSnapshot()
        });

        it('draw an empty header', () => {
            const component = renderer.create(<UserHeader />).toJSON()
            expect(component).toMatchSnapshot()
        });
    });

})
