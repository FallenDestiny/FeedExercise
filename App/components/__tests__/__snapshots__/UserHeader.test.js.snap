// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHeader Rendering draw all the header data with date 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "maxHeight": 55,
      "opacity": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/bruno_mart/128.jpg",
      }
    }
    style={
      Object {
        "aspectRatio": 1,
        "borderRadius": 27.5,
        "height": "100%",
        "resizeMode": "contain",
      }
    }
  />
  <View>
    <Text
      style={
        Object {
          "color": "#151522",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 13,
          "marginLeft": 15,
        }
      }
    >
      Charles Hickle
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#151522",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 13,
            "marginLeft": 15,
          },
          Object {
            "color": "#999999",
          },
        ]
      }
    >
      9 days ago
    </Text>
  </View>
</View>
`;

exports[`UserHeader Rendering draw all the header data without date 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "maxHeight": 55,
      "opacity": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/bruno_mart/128.jpg",
      }
    }
    style={
      Object {
        "aspectRatio": 1,
        "borderRadius": 27.5,
        "height": "100%",
        "resizeMode": "contain",
      }
    }
  />
  <View>
    <Text
      style={
        Object {
          "color": "#151522",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 13,
          "marginLeft": 15,
        }
      }
    >
      Charles Hickle
    </Text>
  </View>
</View>
`;

exports[`UserHeader Rendering draw an empty header 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "maxHeight": 55,
      "opacity": 1,
    }
  }
>
  <View />
</View>
`;
